### PR TITLE
feat(node_exporter): initial implementation

### PR DESCRIPTION
**Pull request by SF2311**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,34 +1,44 @@
...
+| `node_exporter_textfile_dir`        |                           `/var/lib/node_exporter`                            | Directory for the textfile exporter.                                                                                               |
+| `node_exporter_tls_server_config`   |                                     `{}`                                      | Config for the TLS feature.                                                                                                        |
+| `node_exporter_basic_auth_users`    |                                     `{}`                                      | List of users and BCRYPT password hashes for basic auth.                                                                           |
+| `node_exporter_enabled_collectors`  | `["systemd", {"textfile": {"directory": "{{ node_exporter_textfile_dir}}"}}]` | Collectors and their config, that should be enabled regardless of the default.                                                     |
```
Comment (neumantm) [inline README.md]: What does
> regardless of the default

mean here?

File: tasks/main.yml [hunk 2]
```diff
@@ -1,2 +1,95 @@
...
+- name: Set systemd version fact
+  set_fact:
+    node_exporter_systemd_version: "{{ __systemd_version.stdout_lines[0] | regex_replace('^systemd\\s(\\d+).*$', '\\1') }}"
+
+- name: create cache directory
+  file:
+    path: /var/cache/node_exporter/{{ node_exporter_version }}
+    state: directory
+
+- name: Download node_exporter and verify SHA256 checksum
```
Comment (neumantm) [inline tasks/main.yml]:
```suggestion
- name: Download node_exporter
```
I'd say the hash verification is an implementation detail of the download step

File: templates/node_exporter.service.j2 [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+SyslogIdentifier=node_exporter
+Restart=always
+RestartSec=1
+StartLimitInterval=0
+
+{% set protect_home = 'yes' %}
+{% for m in ansible_mounts if m.mount.startswith('/home') %}
+{%   set protect_home = 'read-only' %}
+{% endfor %}
+ProtectHome={{ protect_home }}
```
Comment (neumantm) [inline templates/node_exporter.service.j2]: What exactly does this do and why do we do it?

**Review by neumantm — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (SF2311) [reply README.md]: There are collectors that are available but disabled by default.

File: templates/node_exporter.service.j2 (hunk 3 above)
Comment (SF2311) [reply templates/node_exporter.service.j2]: This renders the service incapable of writing to /home. As discussed we can omit this.

File: README.md (hunk 1 above)
Comment (neumantm) [reply README.md]: Would `This completely overwrites the default of node_exporter.` explain what you mean, or did I misunderstand?

Comment (SF2311) [reply README.md]: On a per collector basis yes. It does not replace the complete list of defaults.

Comment (neumantm) [reply README.md]: Hm it still is not very clear to me what the description here intends to say.
Would the following be correct?
`Collectors that should be enabled (in addition to the node_exporter defaults) and their config (which overwrites any defaults defined by node_exporter).`

Comment (SF2311) [reply README.md]: Maybe the following would be a suitable description:
`Collectors that will be enabled and their respective configuration. In addition to the collectors defined here there may be others, that are enabled by default.`

Comment (neumantm) [reply README.md]: Yes this sounds good

Comment (SF2311) [reply README.md]: Done.

**Review by neumantm — APPROVED** [review]

**Pull request merged**